app/assets/javascripts/discourse/widgets/post-cooked.js

```diff
@@ -41,14 +41,31 @@ export default class PostCooked {
   }
 
   init() {
-    const $html = this._computeCooked();
-    this._insertQuoteControls($html);
-    this._showLinkCounts($html);
-    this._fixImageSizes($html);
-    this._applySearchHighlight($html);
-
-    _decorators.forEach(cb => cb($html, this.decoratorHelper));
-    return $html[0];
+    const cookedDiv = this._computeCooked();
+    const $cookedDiv = $(cookedDiv);
+
+    this._insertQuoteControls($cookedDiv);
+    this._showLinkCounts($cookedDiv);
+    this._fixImageSizes($cookedDiv);
+    this._applySearchHighlight($cookedDiv);
+
+    this._decorateAndAdopt(cookedDiv);
+
+    return cookedDiv;
+  }
+
+  _decorateAndAdopt(cooked) {
+    const $cooked = $(cooked);
+
+    _decorators
```

Comment:
This is going to be called a LOT and the filter is the same result every time.

I would recommend making separate `_beforeAdoptDecorators` and `_afterAdoptDecorators` so you needn't filter every time.

Reply:
Good idea, thanks ❤️